xds/internal/xdsclient/bootstrap/bootstrap.go

```diff
@@ -48,17 +48,53 @@ const (
 	// server supports the v3 version of the xDS transport protocol.
 	serverFeaturesV3 = "xds_v3"
 
-	// Type name for Google default credentials.
-	credsGoogleDefault              = "google_default"
-	credsInsecure                   = "insecure"
 	gRPCUserAgentName               = "gRPC Go"
 	clientFeatureNoOverprovisioning = "envoy.lb.does_not_support_overprovisioning"
 )
 
-var gRPCVersion = fmt.Sprintf("%s %s", gRPCUserAgentName, grpc.Version)
+var (
+	// registry is a map from credential type name to Credential builder.
+	registry = make(map[string]Credentials)
 
-// For overriding in unit tests.
-var bootstrapFileReadFunc = ioutil.ReadFile
+	// For overriding in unit tests.
+	bootstrapFileReadFunc = ioutil.ReadFile
+
+	gRPCVersion = fmt.Sprintf("%s %s", gRPCUserAgentName, grpc.Version)
+)
+
+func init() {
+	RegisterCredentials(&google.DefaultCredsBuilder{})
+	RegisterCredentials(&google.InsecureCredsBuilder{})
+}
```

Comment:
This would go in the `google` / `insecure` packages

Reply:
Moved.